src/software/amazon/ion/impl/SymbolTableReader.java

```diff
@@ -1353,4 +1354,13 @@ public void close() throws IOException
     {
         _current_state = S_EOF;
     }
+
+    public IntegerType getIntegerType()
+    {
+        if (stateType(_current_state) != IonType.INT)
+        {
+            return IntegerType.NONE;
+        }
+        return IntegerType.INT; // all of SymbolTable's integers are type int
```

Comment:
I don't believe the comment.  Please elaborate as to _why_ that is the case.  And perhaps there should be an assertion as well.